libyul/YulStack.cpp

```diff
@@ -117,7 +118,10 @@ void YulStack::optimize()
 
 		m_stackState = Parsed;
 		optimize(*m_parserResult, true);
-		yulAssert(analyzeParsed(), "Invalid source code after optimization.");
```

Comment:
This is what was causing double warnings. `analyzeParsed()` is not side-effect free and actually adds anything that is reported to the current error list. The assumption was probably that a this point nothing will be reported because the code compiles. Warnings and infos are still possible though. They will be reported again unless the code that triggers them happens to be optimized out.